omniledger/service/transaction.go

```diff
@@ -123,25 +105,21 @@ func (instr Instruction) Hash() []byte {
 // InstanceID, the given string, and the hash of the Instruction.
 func (instr Instruction) DeriveID(what string) InstanceID {
 	h := sha256.New()
-	h.Write([]byte(what))
 	h.Write(instr.Hash())
+	h.Write([]byte{0})
```

Comment:
A 0-byte will not protect from any attack, as far as I can see. To prevent attacks we need to add the length of the coming part to be included in the hash, if it is variable-length. So it should be:
```
Write(instr.hash) // fixed length, no protection needed
sigBuf := ... // marshal all signatures
Write(VarInt(len(sigBuf))) // protect against somebody filling up using `what`
Write(sigBuf)
Write(what) // As this is the last element, no need to protect it
```
I hope that is correct. But adding `0` between two `Write` does not solve the problem. Except if we're sure that no `0` shows up in both `Signatures` and `what`, which is not the case, as far as I can tell.

Reply:
This is covered by #1360, so I'm going to ignore this for the moment, in order to get this thing checked in.